### ISSUE TITLE
sphinxext.plot_directive: captions not working

### ISSUE THREAD

**Issue by andrej5elin** [matplotlib/matplotlib#2850]
Dear all,

when including plots in sphinx documents, like:

```
.. plot:: plot.py

   Caption not working!
```

According to documentation, this should include the caption, however
make html will fail because of a RuntimeError("plot:: directive can't have both args and content"), which is raised in line 671 of plot_directive.py.

Removing that line fixes the problem for me, but I wonder why was that restriction implemented in the first place.

**Comment by petehuang** [comment]
What is the recommended action here?

**Comment by aa956** [comment]
I'm successfully using just the following to allow plots to have captions (allows them to be numbered and referenced using numref):
```diff
--- plot_directive.py.original	2017-04-22 16:11:15.897825606 +0300
+++ plot_directive.py	2017-04-22 16:17:58.837148036 +0300
@@ -662,9 +644,6 @@
 
 
 def run(arguments, content, options, state_machine, state, lineno):
-    # The user may provide a filename *or* Python code content, but not both
-    if arguments and content:
-        raise RuntimeError("plot:: directive can't have both args and content")
 
     document = state_machine.document
     config = document.settings.env.config
```

Complete diff (with small changes to template) attached.

[plot_directive.py.diff.txt](https://github.com/matplotlib/matplotlib/files/950651/plot_directive.py.diff.txt)

**Comment by anntzer** [comment]
Closed by #4702 AFAICT, feel free to request a reopen if it's still a problem.
